docs/source/endpoints/upgrade.md

```diff
@@ -0,0 +1,125 @@
+---
+myst:
+  html_meta:
+    "description": "The @upgrade endpoint exposes  Plone backend."
+    "property=og:description": "The @system endpoint exposes system information about the Plone backend."
+    "property=og:title": "Upgrade"
+    "keywords": "Plone, plone.restapi, REST, API, Upgrade"
+---
+
+(upgrade)=
+
+# Upgrade
+
+A Plone site needs to be in sync with the version available on the file system.
+The `@upgrade` endpoint exposes upgrade information about the Plone backend, and supports running the upgrade of the site.
+
+```{note}
+The upgrade endpoint is protected by the `cmf.ManagePortal` permission that requires the Site Administrator or Manager role.
+```
+
+## Get upgrade information
+
+Send a `GET` request to the `@upgrade` endpoint:
+
+```{eval-rst}
+..  http:example:: curl httpie python-requests
+    :request: ../../../src/plone/restapi/tests/http-examples/upgrade_get.req
+```
+
+The response will contain information about available upgrade steps:
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "@id": "http://localhost:55001/plone/@upgrade",
+  "upgrade_steps": {
+    "6006-6007": [
+      {
+        "id": "cb3b910894ab80c94b6367b39bc93039",
+        "title": "Run to6007 upgrade profile."
+      },
+      {
+        "id": "bb72e6a5b0c9c6131b1efad155868131",
+        "title": "Add a timezone property to portal memberdata if it is missing."
+      },
+      {
+        "id": "62ef4eea9b3bfbd472cf349277de749b",
+        "title": "Fix the portal action icons."
+      },
+      {
+        "id": "bac1428c12e294517042a1cee5cfabdc",
+        "title": "Rename the behavior collective.dexteritytextindexer to plone.textindexer"
+      }
+    ],
+    "6007-6008": [
+      {
+        "id": "e5cad1e9fd65e8bd1b23519d49417e51",
+        "title": "Update plonetheme.barceloneta registry"
+      }
+    ]
+  },
+  "versions": {
+    "fs": "6008",
+    "instance": "6006"
+  }
+}
+```
```

Comment:
I thought that the responses are supposed to be in separate files for testing, and referenced like this:

````
```{literalinclude} ../../../src/plone/restapi/tests/http-examples/upgrade_get.resp
:language: http
```
````